### PR TITLE
[WIP] 1 joss new

### PR DESCRIPTION
**Pull request by YipengHu**
# Description

Circulating the JOSS paper draft - please comment / edit.

Fixes # 1

## Type of change

DOC update

### CODE REVIEW

File: docs/joss_paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+`DeepReg` is a Python package that implements a class of image registration algorithms based on deep neural networks. Both unsupervised algorithms [@de2019deep] and weakly-supervised algortihms [@hu2018label] have been implemented. In addition, combining unsupervised losses, such as those derived from image similarity measures, and weak supervision, such as overlap between corresponding anatomical and pathological regions, is at the core of this framework. In this paper, we outline the algorithms and several representative clinical applications, avalibale as 'demos' at the time of writing. An summary of data input and output is also provided to describe the implemented dataset loaders and training data sampling methods. Collectively, we demonstrate the versatility of the package, by providing a wide range of clinical applications as example demonstrations in the `demos` folder.
```
Comment (mathpluscode) [inline docs/joss_paper/paper.md:59]: - typo: avalibale -> available
- An summary -> a summary

File: docs/joss_paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+`DeepReg` is primarily a command line tool that provides basic training and prediction functionalities, with options and configurations being specified though the command line arguments and/or configuration files in .yaml format.
```
Comment (mathpluscode) [inline docs/joss_paper/paper.md:61]: primarily a cli tool? or we also provided a cli tool option?

File: docs/joss_paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+# Algorithms
+## Unsupervised learning
+Unsupervised learning was first developed independently from a number of research groups, among which, [@de2019deep] demonstrated the applicability in real clinical applications while VoxelMorph [@balakrishnan2019voxelmorph] has contributed to a popular open-source repository for several algorithms tested using brain MR images. Image dissimilarity is measured between the fixed and warped moving images, which has been adapted from the classical image registration methods. Fig.\autoref{fig:unsupervised} shows a schematic illustration of the network training. Image dissimilarity measures include sum-of-square difference in intensity (SSD), normalised cross-correlation (NCC), mutual information (MI) and their variants.
```
Comment (mathpluscode) [inline docs/joss_paper/paper.md:65]: Image dissimilarity measures include, here it should be measurements?

File: docs/joss_paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+![Registration network with unsupervised loss.\label{fig:unsupervised}](../asset/deepreg-tutorial-unsupervised.svg)
+
+## Weakly-supervised learning
+Weak supervision utilising segmented corresponding regions in the medical image pairs was first proposed in a multimodal application for registering prostate MR to ultrasound images [@hu2018label]. In addition to the regularisation on the predicted displacement field, the training is driven by minimising the dissimilarity between the fixed labels and the warped moving labels, as illustrated in Fig.X. This formulation is modality-independent and similar to many other well-studied computer vision and medical imaging tasks, such as image segmentation. When multiple corresponding label pairs available, a two-stage sampling strategy can be adopted for an unbiased gradient back-propagation during training. Label dissimilarity measures include Dice, Jaccard, cross-entropy and their variants, such as a multiscale Dice.
```
Comment (mathpluscode) [inline docs/joss_paper/paper.md:70]: same here, measure vs measurement in the last sentence

File: docs/joss_paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+The latest result from a different registration formulation challenged the use of deformation regularisation and consequently the necessity of predicting dense displacement field [@hu2019conditional], also implemented in DeepReg.
+
+# Implemetation
+## Model
+A pair of moving image and fixed image is the input of the network. When available, pairs of moving labels and/or fixed labels can also be the input of the network, regardless of whether they are used to compute the loss. The network predicts a dense displacement field (DDF). Options are also available for the DDF being integrated by a predicted dense velocity field (DFV) or computed from a predicted affine transformation. For conditional segmentation, the network directly predicts warped moving labels [@hu2019conditional].
+
+An encoder-decoder architecture, such as an adapted U-Net and the local-net [@hu2018weakly] is used for dense prediction such as direct DDF or DVF. An encoder-only architecture, such as ResNet and VGG, is used to predict the parameters of an affine (or rigid) transformation model, with 12 (or 6) degrees of freedom.
```
Comment (mathpluscode) [inline docs/joss_paper/paper.md:88]: do we have the two options for 12 and 6 degrees? i have the impression that we only have 12 DOFs

File: docs/joss_paper/paper.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+Unpaired images can also be grouped in applications such as single-modality intra-subject registration, in which each subject has multiple images acquired, e.g. at different time points. Two dataset loaders are implemented for grouped image data, with- and without labels. Paired dataset loader is a special case of the grouped dataset loader. Prostate logitudinal MR is proposed to track the progression of prostate cancer patients at an early stage during the active surveilence programme [@yang2020longitudinal]. One DeepReg demo illustrates how the grouped dataset loader can be used for deep-learning based feature reigstration by aligning intra-patient organ masks.
+
+
+# Conclusion
+`DeepReg` provides a structured code architecture to facilitate a number of deep learning algorithms to train medical image registration, which provides a reference of basic functionality and performance to further research in medical image analysis and computer assisted intervention. It is in its current open-source format not only providing a tool for scientific research, but also welcoming contributions from wider communities above and beyond medical, engineering and computer sicence.
```
Comment (mathpluscode) [inline docs/joss_paper/paper.md:113]: I'd say ... to facilitate the use of a number of ... for image registration.

Also, do we have to strict ourself to "medical image"? i think maybe not

File: docs/joss_paper/paper.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+If images are available in pairs, two dataset loaders are currently available for paired images with- and without labels. Many clinical application for tracking organ motion and other temporal changes between two time points require _intra-subject_ _single-modality_ image registration. Registering lung CT images for the same patient, acquired at expritory and inspritory, is such an example of both unsupervised (without labels) and combined supervision (with segmentation of anatomical structures). Furthermore, registering prostate MR, aquired before surgery, and the intra-operative trasnrectal ultrasound images is an example of weakly-supervised training for multimodal image registration.
+
+## Unpaired images
+Unpaired images are found in applications such as single-modality inter-subject registration. Two dataset loaders for labelled and unlabelled image data. Registering different MR images of brain from different subjects has played a fundamental role in population studies in neuroimaging. An example registering 3D ultrasound images from different surgical patients are also provided, especially to demonstrate the support for experiments such as cross-validation in DeepReg.
```
Comment (mathpluscode) [inline docs/joss_paper/paper.md:106]: - "Two dataset loaders for labelled and unlabelled image data." this sentence is incomplete
- Last sentence, An example registering ... "are" also provided, should be "is"

File: docs/joss_paper/paper.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+## Grouped images
+Unpaired images can also be grouped in applications such as single-modality intra-subject registration, in which each subject has multiple images acquired, e.g. at different time points. Two dataset loaders are implemented for grouped image data, with- and without labels. Paired dataset loader is a special case of the grouped dataset loader. Prostate logitudinal MR is proposed to track the progression of prostate cancer patients at an early stage during the active surveilence programme [@yang2020longitudinal]. One DeepReg demo illustrates how the grouped dataset loader can be used for deep-learning based feature reigstration by aligning intra-patient organ masks.
```
Comment (mathpluscode) [inline docs/joss_paper/paper.md:109]: so the grouped image demo has to be ready before the submission?

File: docs/joss_paper/paper.md (hunk 3 above)
Comment (YipengHu) [reply docs/joss_paper/paper.md:65]: this is a sort of a term "dissimilarity measures"

File: docs/joss_paper/paper.md (hunk 6 above)
Comment (YipengHu) [reply docs/joss_paper/paper.md:113]: all the demos are medical imaging and hard to see the 3d images outside of this area...

File: docs/joss_paper/paper.md (hunk 8 above)
Comment (YipengHu) [reply docs/joss_paper/paper.md:109]: yes

**Comment by YipengHu** [comment]
@mathpluscode having difficulty to review the paper using this: https://whedon.theoj.org/
Can you have a look?

**Pull request merged**